pubber/bin/run

```diff
@@ -2,10 +2,19 @@
 
 ROOT_DIR=$(realpath $(dirname $0)/../..)
 
-java -Dorg.slf4j.simpleLogger.showThreadName=false \
-     -jar $ROOT_DIR/pubber/build/libs/pubber-1.0-SNAPSHOT-all.jar \
-     $@ &
+result=192  # Exit code used to indicate intentional system restart.
 
-echo $! > $ROOT_DIR/.pubber.pid
+while [[ $result -eq 192 ]]; do
+    java -Dorg.slf4j.simpleLogger.showThreadName=false \
+         -jar $ROOT_DIR/pubber/build/libs/pubber-1.0-SNAPSHOT-all.jar \
+         $@ &
 
-wait
+    PUB_PID=$!
+
+    result=0
+    echo Waiting for pubber pid $PUB_PID to complete...
+    wait $PUB_PID || result=$?
+done
+
+echo Done with pubber run, exit code $result
```

Comment:
If wait fails we won't know it versus the process exiting with status 0. Do you care? 
If so you could assign wait $PUB_PID; result=$?   instead of ||   and get either wait failure 127 or the exit status.